Utils/src/BTaggingEffAnalyzer.cc

```diff
@@ -0,0 +1,233 @@
+      TH2D  *h2_BTaggingEff_Num_b;
+      TH2D  *h2_BTaggingEff_Num_c;
+      TH2D  *h2_BTaggingEff_Num_udsg;
+};
+
+//
+// constants, enums and typedefs
+//
+typedef std::vector<pat::Jet> PatJetCollection;
+
+//
+// static data member definitions
+//
+
+//
+// constructors and destructor
+//
+BTaggingEffAnalyzer::BTaggingEffAnalyzer(const edm::ParameterSet& iConfig) :
+
+  jetsTag(iConfig.getParameter<edm::InputTag>("JetsTag")),
+  discriminatorTag(iConfig.getParameter<std::string>("DiscriminatorTag")),
+  discriminatorValue(iConfig.getParameter<double>("DiscriminatorValue")),
+  ptNBins(iConfig.getParameter<int>("PtNBins")),
+  ptMin(iConfig.getParameter<double>("PtMin")),
+  ptMax(iConfig.getParameter<double>("PtMax")),
+  etaNBins(iConfig.getParameter<int>("EtaNBins")),
+  etaMin(iConfig.getParameter<double>("EtaMin")),
+  etaMax(iConfig.getParameter<double>("EtaMax"))
+
+{
+   //now do what ever initialization is needed
+   h2_BTaggingEff_Denom_b    = fs->make<TH2D>("h2_BTaggingEff_Denom_b", ";p_{T} [GeV];#eta", ptNBins, ptMin, ptMax, etaNBins, etaMin, etaMax);
+   h2_BTaggingEff_Denom_c    = fs->make<TH2D>("h2_BTaggingEff_Denom_c", ";p_{T} [GeV];#eta", ptNBins, ptMin, ptMax, etaNBins, etaMin, etaMax);
+   h2_BTaggingEff_Denom_udsg = fs->make<TH2D>("h2_BTaggingEff_Denom_udsg", ";p_{T} [GeV];#eta", ptNBins, ptMin, ptMax, etaNBins, etaMin, etaMax);
+   h2_BTaggingEff_Num_b    = fs->make<TH2D>("h2_BTaggingEff_Num_b", ";p_{T} [GeV];#eta", ptNBins, ptMin, ptMax, etaNBins, etaMin, etaMax);
+   h2_BTaggingEff_Num_c    = fs->make<TH2D>("h2_BTaggingEff_Num_c", ";p_{T} [GeV];#eta", ptNBins, ptMin, ptMax, etaNBins, etaMin, etaMax);
+   h2_BTaggingEff_Num_udsg = fs->make<TH2D>("h2_BTaggingEff_Num_udsg", ";p_{T} [GeV];#eta", ptNBins, ptMin, ptMax, etaNBins, etaMin, etaMax);
+}
+
+BTaggingEffAnalyzer::~BTaggingEffAnalyzer()
+{
+ 
+   // do anything here that needs to be done at desctruction time
+   // (e.g. close files, deallocate resources etc.)
+
+}
+
+
+//
+// member functions
+//
+
+// ------------ method called for each event  ------------
+void
+BTaggingEffAnalyzer::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetup)
+{
+  edm::Handle<PatJetCollection> jets;
+  iEvent.getByLabel(jetsTag,jets);
+
+  //grab event selection variables:
+  edm::Handle<double> HT;
+  edm::Handle<double> MHT; 
+  edm::Handle<double>DeltaPhi1;
+  edm::Handle<double>DeltaPhi2;
+  edm::Handle<double>DeltaPhi3;
+  edm::Handle<double>DeltaPhi4;
+  edm::Handle<int> NJets;
+  edm::Handle<int> Leptons;
+  edm::Handle<int> isoPionTracks;
+  edm::Handle<int> isoMuonTracks;
+  edm::Handle<int> isoElectronTracks;
+  edm::Handle<bool> JetID; 
+
+  iEvent.getByLabel("HT", HT);
+  iEvent.getByLabel("MHT","Pt", MHT);
+  iEvent.getByLabel("DeltaPhi", "DeltaPhi1", DeltaPhi1);
+  iEvent.getByLabel("DeltaPhi","DeltaPhi2", DeltaPhi2);
+  iEvent.getByLabel("DeltaPhi","DeltaPhi3", DeltaPhi3);
+//  iEvent.getByLabel("DeltaPhi","DeltaPhi4", DeltaPhi4);
```

Comment:
You don't have to make a new pull request when you rebase. You may have to force push to origin because the rebase rewrites history:
```
git push -f origin
```
but then it will automatically update this PR.